functionary/core/utils/parameter.py

```diff
@@ -96,6 +96,7 @@ def validate_parameters(parameters: dict, instance: Union["Function", "Workflow"
     """
     try:
         pydantic_model = _get_pydantic_model(instance)
+        # KeyError will get thrown if one or more parameters are invalid
```

Comment:
I'm pretty sure this is actually a bug that needs to be fixed rather than just catching the `KeyError`.  `_serialize_json_parameters` assumes that all of the JSON parameters that a function takes are present in the parameters input, which is obviously not always going to be the case.  It should have `name__in=parameters.keys()` added to the filter so that it's only attempting to serialize the parameters that are actually present.  Then you would end up with a `ValidationError` as desired, rather than this `KeyError`.

Side note: we should absolutely be writing unit tests for situations like this.  We write a test to reproduce the bug, fix the bug, see the test pass.  This is technically not related to the thing that this PR is about, so if you want to take this change out and open a separate ticket for this bug I'm okay with that.  In fact, maybe it'd be better to do that because I actually want to talk over some of what's going on here with the model validation as I don't particularly love certain aspects of it.